### DISCUSSION TITLE
(New) Casadi MX.sym() derivative result interpretation has become cryptic

### DISCUSSION THREAD

**Discussion by MariosGkMeng** [casadi/casadi#3459] [Q&A]
Hello all,

It's been 1.5 years since I have used Casadi.

Today I need it just to check some mathematical expressions, using the MX.sym(). 

I am performing some matrix derivations using the `jacobian` function.

I am **not able to interpret the results**, and I did not find anything in the documentation. 
That output is:

>> simple.D_E_t

ans = 

\@\1=(mac(W_t_simple,X,zeros(2x15))-mac(W_simple,X,zeros(2x15))), 
\@\2=(-mac(ones(2x2,1nz),X,zeros(2x15))), 
\@\3=\@\1', 
\@\4=(-mac(ones(2x2,1nz),X,zeros(2x15))), 
\@\5=(-mac(ones(2x2,1nz),X,zeros(2x15))), 
\@\6=(-mac(ones(2x2,1nz),X,zeros(2x15))), 

THE LINE THAT CONFUSES ME:
((((zeros(4x4,12nz)[0, 2, 6] = (mac(\@\1,\@\2',zeros(2x2))+mac(\@\2,\@\3,zeros(2x2)))[:3])[3, 7, 10] = (mac(\@\1,\@\4',zeros(2x2))+mac(\@\4,\@\3,zeros(2x2)))[1:4])[1, 4, 8] = (mac(\@\1,\@\5',zeros(2x2))+mac(\@\5,\@\3,zeros(2x2)))[:3])[5, 9, 11] = (mac(\@\1,\@\6',zeros(2x2))+mac(\@\6,\@\3,zeros(2x2)))[1:4])'

How am I supposed to read the result above?
I remember the entries separated by commas, but this is very peculiar. 

Thank you in advance

**Comment by jaeandersson** [comment]
Try to generate c code. The result will be more verbose but more readable and better maps to the actual expressions.

**Reply by MariosGkMeng** [reply]
Thank you for the reply! 

I will try that
